packages/filesystem/src/electron-browser/file-dialog/electron-file-dialog-service.ts

```diff
@@ -49,7 +49,11 @@ export class ElectronFileDialogService extends DefaultFileDialogService {
     override async showOpenDialog(props: OpenFileDialogProps, folder?: FileStat): Promise<MaybeArray<URI> | undefined> {
         const rootNode = await this.getRootNode(folder);
         if (rootNode) {
-            const { filePaths } = await electronRemote.dialog.showOpenDialog(this.toOpenDialogOptions(rootNode.uri, props));
+            const electronWindow = electronRemote.getCurrentWindow();
+            const isModal = props.modal !== false;
+            const { filePaths } = isModal ?
+                await electronRemote.dialog.showOpenDialog(electronWindow, this.toOpenDialogOptions(rootNode.uri, props)) :
+                await electronRemote.dialog.showOpenDialog(this.toOpenDialogOptions(rootNode.uri, props));
```

Comment:
minor: we can simplify the expression to only get the current window if necessary:

```suggestion
            const { filePaths } = props.modal !== false ?
                await electronRemote.dialog.showOpenDialog(electronRemote.getCurrentWindow(), this.toOpenDialogOptions(rootNode.uri, props)) :
                await electronRemote.dialog.showOpenDialog(this.toOpenDialogOptions(rootNode.uri, props));
```